maidroid_core/cores/ocr.lua

```diff
@@ -327,19 +325,42 @@ local maidroid_instruction_set = {
 			under = under,
 			type = "node"
 		}
-		local newitem, succ = stack:get_definition().on_place(stack, obj, pt)
+		local stack_def = stack:get_definition()
+		local newitem, succ = stack_def.on_place(stack, thread.droid.object, pt)
 		if not succ then
 			return true, false, "could not place"
 		end
 
+		-- play the place sound
+		if stack_def.sounds and stack_def.sounds.place then
+			minetest.sound_play(stack_def.sounds.place, {
+				pos = pos,
+				max_hear_distance = 10,
+			})
+		end
+
 		-- set new item
 		if newitem then
-			obj:get_wielded_item(newitem)
+			thread.droid:set_wielded_item(newitem)
 		end
 
 		return true, true
 	end,
 
+	select_item = function(params, thread)
+		if not params[1] or type(params[1]) ~= "number" then
+			return false, "number expected"
+		end
+		local inv = thread.droid:get_inventory()
+		if params[1] < 1 or params[1] > inv:get_size("main") then
```

Comment:
It doesn't have to be an integer, `inv:g/set_stack` works with any number (tested).

So, add `params[1] == NaN`?